op-supervisor/supervisor/backend/cross/cycle_test.go

```diff
@@ -527,6 +529,54 @@ func TestHazardCycleChecksCycle(t *testing.T) {
 			},
 			msg: "expected cycle detection error for cycle through executing messages",
 		},
+		{
+			// 0->1->0
+			name: "cycle through single chain, exec message prior to init and adjacent",
+			chainBlocks: map[string]chainBlockDef{
+				"1": {
+					logCount: 2,
+					messages: map[uint32]*types.ExecutingMessage{
+						0: execMsg("1", 1),
+					},
+				},
+			},
+			expectErr: ErrCycle,
+			msg:       "expected cycle detection error",
+		},
+		{
+			// 0->1->2->0
```

Comment:
```suggestion
			// 0->2->1->0
```
(Assuming `exec -> init` format, the other way around is fine too, but need some consistency across comments)